chargeRateStructs.cpp

```diff
@@ -0,0 +1,25 @@
+#include "chargeRateStructs.h"
+
+const char* delimiter = ",";
+const int conversionBase = 10; 
+std::list<long> sessionValues = {};
+std::list<chargeRange> chargeRanges = {};
+
+bool chargeRange::calculateReadings(std::list<long> &samples){
+    for(std::list<long>::iterator sampleIt = samples.begin(); 
+        sampleIt != samples.end(); sampleIt++){
+        if((*sampleIt >= lowValue) && (*sampleIt <= highValue)){
+            samples.erase(sampleIt);
```

Comment:
@FernandoRiv you are erasing the iterator item while iterating the list.
That is why I think it's giving a segmentation violation. It happened for me while compiling and running using WSL (ubuntu) in Windows.
In general, it's not a good practice to erase items during iteration of the items. Try working out the problem in smaller steps and look for a simpler solution

Reply:
Got it, thanks Sudeep! I've suggested a new proposal now :)